File: lib/models/appModels/JoinConditions.dart
import 'Event.dart';
import 'Team.dart';

class JoinConditions {
  bool? withRequest;
  bool? withPayment;
  bool? forTeam;
  bool? forEvent;
  Event? event;
  Team? team;

  JoinConditions({
    this.withRequest,
    this.withPayment,
    this.forTeam,
    this.forEvent,
    this.event,
    this.team,
  });
}

File: apollo/src/JoinConditions.ts
import mongoose from 'mongoose';



const Schema = mongoose.Schema;


const joinConditionsSchema = new mongoose.Schema({
    withRequest: Boolean,
    withPayment: Boolean,
    forTeam: Boolean,
    forEvent: Boolean,
    event: { type: Schema.Types.ObjectId, ref: 'Event' },
});

const JoinConditions = mongoose.model('JoinConditions', joinConditionsSchema);

export default JoinConditions;